pkg/process/net/common.go

```diff
@@ -129,69 +129,77 @@ func (r *RemoteSysProbeUtil) GetProcStats(pids []int32) (*model.ProcStatsWithPer
 	return results, nil
 }
 
-// GetConnections returns a set of active network connections, retrieved from the system probe service, supports the
-// grpc server as well as the http server.
-func (r *RemoteSysProbeUtil) GetConnections(clientID string) (*model.Connections, error) {
-	var conns *model.Connections
-	useGRPCServer := dd_config.SystemProbe.GetBool("service_monitoring_config.use_grpc")
-	if useGRPCServer {
-		conn, err := grpc.Dial("unix:///tmp/my_grpc.sock", grpc.WithInsecure())
-		if err != nil {
-			log.Errorf("Failed to connect: %v", err)
-		}
+func (r *RemoteSysProbeUtil) getConnectionWithRPC(unixSockPath, clientID string) (*model.Connections, error) {
+	conn, err := grpc.Dial(unixSockPath, grpc.WithInsecure())
+	if err != nil {
+		return nil, err
+	}
+
+	client := connectionserver.NewSystemProbeClient(conn)
 
-		client := test2.NewSystemProbeClient(conn)
+	response, err := client.GetConnections(context.Background(), &connectionserver.GetConnectionsRequest{ClientID: clientID})
+	if err != nil {
+		return nil, err
+	}
 
-		response, err := client.GetConnections(context.Background(), &test2.GetConnectionsRequest{ClientID: clientID})
-		if err != nil {
-			log.Errorf("Failed to call get connections: %v", err)
-		}
+	res, err := response.Recv()
+	if err != nil {
+		return nil, err
+	}
 
-		res, err := response.Recv()
-		if err != nil {
-			log.Errorf("Failed to get response: %v", err)
-		}
+	conns, err := netEncoding.GetUnmarshaler("application/protobuf").Unmarshal(res.Data)
+	if err != nil {
+		return nil, err
+	}
+	fmt.Printf("[GetConnections-rpc] we are in the grpc mode, found %d conns with the grpc server\n", len(conns.Conns))
```

Comment:
is it a temporary print? or do you intend to keep it?

Reply:
yes